dotnet/src/InternalUtilities/src/Data/VectorStoreRecordPropertyReader.cs

```
@@ -337,6 +338,76 @@ public string GetJsonPropertyName(string dataModelPropertyName)
         return this._jsonPropertyNamesMap.Value[dataModelPropertyName];
     }
 
+    /// <summary>
+    /// Get the <see cref="VectorStoreRecordProperty.DataModelPropertyName"/> for the given search options.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">When there is no match.</exception>
+    public string GetVectorPropertyName<TRecord>(VectorSearchOptions<TRecord>? searchOptions)
+    {
+        if (searchOptions is not null)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            string? obsoletePropertyName = searchOptions.VectorPropertyName;
+#pragma warning restore CS0618 // Type or member is obsolete
+            if (!string.IsNullOrEmpty(obsoletePropertyName))
+            {
+                return obsoletePropertyName!;
```

We are missing some validation here, to ensure that the name is actually referring to a vector property.
In some connectors I see we are subsequently calling GetProperty, but it would only check that the property exists, not that it is a vector.  Not checking upfront might lead to poor error messaging later if we just pass the name down to the SDK and let it throw.

In my branch I'm adding a propertyreader method that deals with this checking, so we could hold off until mine is merged and then use that here too.